hyrax/templates/NOTES.txt

```diff
@@ -0,0 +1,21 @@
+1. Get the application URL by running these commands:
+{{- if .Values.ingress.enabled }}
+{{- range $host := .Values.ingress.hosts }}
+  {{- range .paths }}
+  http{{ if $.Values.ingress.tls }}s{{ end }}://{{ $host.host }}
+  {{- end }}
+{{- end }}
+{{- else if contains "NodePort" .Values.service.type }}
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ include "hyrax.fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "hyrax.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "hyrax.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "hyrax.name" . }},app.kubernetes.io/instance={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl --namespace {{ .Release.Namespace }} port-forward $POD_NAME 8080:3000
+{{- end }}
```

Comment:
this directory is the helm chart from Hyrax, modified to use mysql instead of postgres. Helm chart overviews can found here: https://helm.sh, https://www.freecodecamp.org/news/what-is-a-helm-chart-tutorial-for-kubernetes-beginners/ 


Big broad strokes is that the template files get combined with the passend in values from ENV.tmpl.yaml (in ops) to produce the yaml files that are sent to the K8S cluster and declare state. This is similar to how a Rails view is a template that has variables filled in to create an HTML file. 

So to deploy the worker you need to create a deployment-worker.yaml file that has the all values for that environment filled in to it (to know where solr and redis are for example, and to have the db password). Helm does the templeting and creates versions of the rolled out yaml files which allows for roll backs.